app/src/main/java/com/resocoder/forecastmvvm/data/repository/ForecastRepositoryImpl.kt

```diff
@@ -50,7 +53,6 @@ class ForecastRepositoryImpl(
     }
 
     private suspend fun initWeatherData() {
-        val lastWeatherLocation = weatherLocationDao.getLocation().value
 
         if (lastWeatherLocation == null
             || locationProvider.hasLocationChanged(lastWeatherLocation)) {
```

Comment:
Since lastWeatherLocation is nullable, it's impossible smart cast that mutable property to WeatherLocation. As we're sure it cannot be null at this point, it's possible to implement non-null call as to **lastWeatherLocation!!** Same logic applies to isFetchCurrentNeeded(**lastWeatherLocation!!**.zonedDateTime)